extensions/json-schema/services/src/main/java/org/apache/unomi/schema/impl/SchemaServiceImpl.java

```diff
@@ -111,12 +112,52 @@ public boolean isEventValid(String event) {
 
     @Override
     public Set<ValidationError> validateEvent(String event) throws ValidationException {
-        JsonNode jsonEvent = parseData(event);
-        String eventType = extractEventType(jsonEvent);
+        return validateNodeEvent(parseData(event));
+    }
+
+    @Override
+    public Map<String, Set<ValidationError>> validateEvents(String events) throws ValidationException {
+        Map<String, Set<ValidationError>> errorsPerEventType = new HashMap<>();
+        JsonNode eventsNodes = parseData(events);
+        eventsNodes.forEach(event -> {
+            String eventType = event.get("eventType").asText();
```

Comment:
Could be interesting to make the eventType part of the ValidationException to make it a bit more generic.
This would allow to avoid parsing the event type here, as it is supposed to be done in the **extractEventType** function.